services/121-service/src/payments/fsp-integration/safaricom/safaricom.service.ts

```diff
@@ -157,13 +95,13 @@ export class SafaricomService
       InitiatorName: process.env.SAFARICOM_INITIATORNAME!,
       SecurityCredential: process.env.SAFARICOM_SECURITY_CREDENTIAL!,
       CommandID: 'BusinessPayment',
-      Amount: paymentData.transactionAmount,
+      Amount: transactionAmount,
       PartyA: process.env.SAFARICOM_PARTY_A!,
-      PartyB: paymentData.paymentAddress,
+      PartyB: phoneNumber, // Set to empty string to trigger mock failure
       Remarks: `Payment ${paymentNr}`,
       QueueTimeOutURL: EXTERNAL_API.safaricomQueueTimeoutUrl,
       ResultURL: EXTERNAL_API.safaricomResultUrl,
-      Occassion: paymentData.referenceId,
+      Occassion: referenceId,
```

Comment:
The branch that you threw away had relevant comments about what to do with some of the params set here. I suggest to dig these comments up, so we do not need to reinvent the wheel.

And leave in the comments about the "what and why", as a piece of documentation for our future selves.

Reply:
@diderikvw I did include these comments based on your input before I threw it away. Wasn't in this branch yet.